backend/src/entities/Edge.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Represents an edge / a relations of a graph.
+ */
+export interface Edge {
```

Comment:
Maybe rename this to `EdgeDescriptor`, as this has the same role as a pure id has for a node.